streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -892,12 +892,14 @@ object StreamingContext extends Logging {
   }
 
   private[streaming] def rddToFileName[T](prefix: String, suffix: String, time: Time): String = {
-    if (prefix == null) {
-      time.milliseconds.toString
-    } else if (suffix == null || suffix.length ==0) {
-      prefix + "-" + time.milliseconds
-    } else {
-      prefix + "-" + time.milliseconds + "." + suffix
+    var result = new StringBuilder()
+    if (prefix != null && prefix.length > 0) {
+      result = result.append(prefix).append("-")
+    }
+    result = result.append(time.milliseconds)
+    if (suffix != null && suffix.length > 0) {
+      result = result.append(".").append(suffix)
     }
+    result.toString()
```

Comment:
I think it's fine; it might be simpler without bothering with `StringBuilder` but not by much.
(In your code you should use `val result` and you do not need to assign `result` each time.)

```
var result = time.milliseconds.toString
if (prefix != null && prefix.length > 0) {
  result = prefix + "-" result
}
if (suffix != null && suffix.length > 0) {
  result = result + "." + suffix
}
result
```